src/App.tsx

```diff
@@ -0,0 +1,177 @@
+import * as React from 'react';
+import { CheckBox } from 'react-native-elements';
+import {
+  StyleSheet,
+  View,
+  TextInput,
+  Keyboard,
+  Platform,
+  AsyncStorage,
+  FlatList,
+  Button
+} from "react-native";
+
+const isAndroid = Platform.OS == "android";
+const viewPadding = 10;
+
+type ITaskProps = Array<{ key: string, text: string, isCompleted?: boolean }>;
+
+export interface ITodoListState {
+  tasks: ITaskProps;
+  text: string;
+  viewPadding?: string | number;
+}
+
+export default class App extends React.Component<{}, ITodoListState> {
+
+  state = {
+    tasks: [],
+    text: "",
+    viewPadding: viewPadding
+  }
+
+  async componentDidMount() {
+    Keyboard.addListener(
+      isAndroid ? "keyboardDidShow" : "keyboardWillShow",
+      e => this.setState({ viewPadding: e.endCoordinates.height + viewPadding })
+    );
+
+    Keyboard.addListener(
+      isAndroid ? "keyboardDidHide" : "keyboardWillHide",
+      () => this.setState({ viewPadding: viewPadding })
+    );
+
+    const tasks = await this.getTasksFromStorage()
+    this.setState({tasks});
+  }
+
+  addTask = () => {
+    if (this.state.text.trim().length > 0) {
+      this.setState(
+        (prevState: ITodoListState) => {
+          let { tasks, text } = prevState;
+          const currentTaskIndex = tasks.length ? Number(tasks[tasks.length - 1].key) + 1 : 0;
+          return {
+            tasks: tasks.concat({ key: String(currentTaskIndex), text: text }),
+            text: ""
+          };
+        },
+        () => AsyncStorage.setItem("TASKS", JSON.stringify(this.state.tasks))
+      );
+    }
+  };
+
+  deleteTask = (index: number) => {
+    this.setState(
+      (prevState: ITodoListState) => {
+        let tasks = prevState.tasks.slice();
+
+        tasks.splice(index, 1);
+
+        return { tasks: tasks };
+      },
+      () => AsyncStorage.setItem("TASKS", JSON.stringify(this.state.tasks))
+    );
+  };
+
+  toggleStatus = (index: number) => {
+    this.setState(
+      (prevState: ITodoListState) => {
+        let updatedTasks = prevState.tasks.map((task, taskIndex) => {
+          if (taskIndex === index) {
+            return { ...task, isCompleted: !task.isCompleted }
+          }
+
+          return task;
+        });
+
+        return { tasks: updatedTasks };
+      }
+    )
+  }
+
+  async getTasksFromStorage() {
+    const items = await AsyncStorage.getItem("TASKS");
+
+    return items && items.length ? JSON.parse(items) : [];
+  }
+
+  render(): JSX.Element {
+    return (
+      <View style={[styles.container, { paddingBottom: this.state.viewPadding }]}>
+        <FlatList
```

Comment:
list and text inputs can be broken down into shorter renders.

Reply:
That would look weird. Since this view only renders `FlatList` and `TextInput` and there is no logic. I am just passing the props. 
`renderItem` prop's implementation has been moved to a separate function. That should make it readable now.